### PR TITLE
Remove Type Casts

### PR DESCRIPTION
**Pull request by mt-xing**
### Summary 

Standardize types for everything.
All ship angles, health, and times are `float`.
All remaining integral types should be `uint8_t` - 8 bit unsigned integers.
Note these are capped at 255, but we really shouldn't have more than 255 breaches or doors anyways.

### Testing 

Runs on Windows

### Review Focus 

Please check anything that you wrote.

### CODE REVIEW

File: source/StabilizerModel.cpp [hunk 1]
```diff
@@ -7,7 +7,7 @@ constexpr unsigned int SUCCESS_CUTOFF = 60;
 
 StabilizerModel::StabilizerModel()
 	: currState(StabilizerModel::StabilizerState::Inactive), progress(0), endTime(0) {
-	srand(time(nullptr));
+	srand((unsigned int)time(nullptr));
```
Comment (Jeffrey-J-Yao) [inline source/StabilizerModel.cpp:10]: Does this need to be changed to `uint8_t`? Not sure if it's passed around through network synchronization or not.

**Review by Jeffrey-J-Yao — COMMENTED** [review]
lgtm for the most part, but not sure about `StabilizerModel.cpp`

File: source/StabilizerModel.cpp (hunk 1 above)
Comment (mt-xing) [reply source/StabilizerModel.cpp:10]: Nah, that's just used as a seed for the local RNG.

**Review by Jeffrey-J-Yao — APPROVED** [review]
lgtm

**Pull request merged**